src/addon/manager_ui.cpp

```diff
@@ -270,20 +270,28 @@ bool ad_hoc_addon_fetch_session(const std::vector<std::string>& addon_ids)
 		}
 
 		bool return_value = true;
+		std::ostringstream os;
 		for(const std::string & addon_id : addon_ids) {
 			addons_list::const_iterator it = addons.find(addon_id);
 			if(it != addons.end()) {
 				const addon_info& addon = it->second;
 				addons_client::install_result res = client.install_addon_with_checks(addons, addon);
 				return_value = return_value && (res.outcome == addons_client::install_outcome::success);
 			} else {
-				utils::string_map symbols;
-				symbols["addon_id"] = addon_id;
-				gui2::show_error_message(VGETTEXT("Could not find an add-on matching id $addon_id on the add-on server.", symbols));
+				if(!return_value) {
+					os << ", ";
+				}
+				os << addon_id;
 				return_value = false;
 			}
 		}
 
+		if(!return_value) {
+			utils::string_map symbols;
+			symbols["addon_id"] = os.str();
+			gui2::show_error_message(VGETTEXT("Could not find an add-on matching the ids $addon_id on the add-on server.", symbols));
```

Comment:
There's a mix of singular and plural here.

Reply:
Updated.